compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -3478,7 +3478,7 @@ object Types {
       if (selfTypeCache == null)
         selfTypeCache = {
           val given = cls.givenSelfType
-          if (!given.exists) appliedRef
+          if (!given.isValueType) appliedRef
```

Comment:
Might be worth explicitly checking that an error happened like other commits in this PR do, e.g.:
```scala
if (!given.isValueType) {
  assert(!given.exists || ctx.reporter.errorsReported)
  appliedRef
}
```

Reply:
The given type might be higher-kinded (in fact that's what it was in the test) but this would be detected only later.

Reply:
OK, in that case maybe just add a comment explaining that non-value types self can only happen with erroneous code? Otherwise readers of this code might be puzzled.